telegram/ext/filters.py

```diff
@@ -372,1895 +363,1807 @@ def name(self, name: str) -> NoReturn:
+                )
+
+            parsed_username = self._parse_username(username)
+            self._usernames -= parsed_username
+
+    def remove_chat_ids(self, chat_id: SLT[int]) -> None:
+        with self.__lock:
+            if self._usernames:
+                raise RuntimeError(
+                    f"Can't set {self.chat_id_name} in conjunction with (already set) "
+                    f"{self.username_name}s."
+                )
+            parsed_chat_id = self._parse_chat_id(chat_id)
+            self._chat_ids -= parsed_chat_id
+
+    def filter(self, message: Message) -> bool:
+        chat_or_user = self.get_chat_or_user(message)
+        if chat_or_user:
+            if self.chat_ids:
+                return chat_or_user.id in self.chat_ids
+            if self.usernames:
+                return bool(chat_or_user.username and chat_or_user.username in self.usernames)
+            return self.allow_empty
+        return False
+
+    @property
+    def name(self) -> str:
+        return (
+            f'filters.{self.__class__.__name__}('
+            f'{", ".join(str(s) for s in (self.usernames or self.chat_ids))})'
+        )
+
+    @name.setter
+    def name(self, name: str) -> NoReturn:
+        raise RuntimeError(f'Cannot set name for filters.{self.__class__.__name__}')
 
-    successful_payment = _SuccessfulPayment()
-    """Messages that confirm a :class:`telegram.SuccessfulPayment`."""
 
-    class _PassportData(MessageFilter):
+class User(_ChatUserBaseFilter):
+    # pylint: disable=useless-super-delegation
+    """Filters messages to allow only those which are from specified user ID(s) or
+    username(s).
+
+    Examples:
+        ``MessageHandler(filters.User(1234), callback_method)``
+
+    Warning:
+        :attr:`user_ids` will give a *copy* of the saved user ids as :class:`frozenset`. This
+        is to ensure thread safety. To add/remove a user, you should use :meth:`add_usernames`,
+        :meth:`add_user_ids`, :meth:`remove_usernames` and :meth:`remove_user_ids`. Only update
+        the entire set by ``filter.user_ids/usernames = new_set``, if you are entirely sure
+        that it is not causing race conditions, as this will complete replace the current set
+        of allowed users.
+
+    Args:
+        user_id(:obj:`int` | Tuple[:obj:`int`] | List[:obj:`int`], optional):
+            Which user ID(s) to allow through.
+        username(:obj:`str` | Tuple[:obj:`str`] | List[:obj:`str`], optional):
+            Which username(s) to allow through. Leading ``'@'`` s in usernames will be
+            discarded.
+        allow_empty(:obj:`bool`, optional): Whether updates should be processed, if no user
+            is specified in :attr:`user_ids` and :attr:`usernames`. Defaults to :obj:`False`.
+
+    Attributes:
+        user_ids (set(:obj:`int`)): Which user ID(s) to allow through.
+        usernames (set(:obj:`str`)): Which username(s) (without leading ``'@'``) to allow through.
```

Comment:
> Can we make this docstrings of the actual properties?

Um for `user_ids` we can, but for `usernames` we can't since that is defined in the super class and that is private (same for `allow_empty`). Also I think it makes more sense to document them here instead of in the actual property. 

Agreed on the warning thing tho

Reply:
> Um for `user_ids` we can, but for `usernames` we can't since that is defined in the super class and that is private (same for `allow_empty`).

We probably could, if we override the property to call `super().usernames` 🤔 - bascially what we do for the `add/remove_*` methods

> Also I think it makes more sense to document them here instead of in the actual property.

Your call :)